### PR TITLE
feature: add loader for embed.FS

### PR DESCRIPTION
**Pull request by fahmifan**
Support loading template from Go `embed.FS`

To support `embed.FS` we need to upgrade the go module version to at least `v1.16`

There are also some limitations for the filename, please check:
- https://pkg.go.dev/embed#hdr-Directives
- https://github.com/golang/go/issues/43854

### CODE REVIEW

**Comment by whs** [comment]
I wrote my own before seeing this PR, but mine use fs.FS.

I'd suggest that the loader operate on fs.FS, which embed.FS implements. This would allow any alternative filesystem implementation to be used. It probably even allow current LocalFilesystemLoader to be a shim to `os.DirFS`, and HttpFilesystemLoader should also satisfy `fs.FS` (however it require Go 1.16 and break backwards compatibility).

**Comment by fahmifan** [comment]
@whs Oh I see, using into interface would make it more flexible. I'll try to look into it later :+1:

File: template_loader.go [hunk 1]
```diff
@@ -211,3 +212,56 @@ func (h *HttpFilesystemLoader) Get(path string) (io.Reader, error) {
...
+}
+
+// Abs return the join of base and name
+func (e *EmbededLoader) Abs(base, name string) string {
+	return filepath.Join(filepath.Dir(base), name)
+}
+
+// Get returns an io.Reader where the template's content can be read from.
+func (e *EmbededLoader) Get(path string) (reader io.Reader, err error) {
+	f, err := e.fs.Open(path)
```
Comment (whs) [inline template_loader.go:255]: e.fs.Open return type already satisfy (io.Reader, err), any reason why this read the file into memory?

Comment (fahmifan) [reply template_loader.go:255]: it was open a file, so the caller should responsible to close them. Unless, the `Get` method return signature is `io.ReadCloser`

**Comment by iredmail** [comment]
This PR breaks backward compatibility which requires Go 1.16. I suggest check with @flosch first to make sure this is acceptable. :)

File: template_loader.go (hunk 1 above)
Comment (whs) [reply template_loader.go:255]: Good point, I didn't realize it wasn't ReadCloser

**Comment by fahmifan** [comment]
@flosch hey, i already add the unit test and as mentioned in the comment above this PR will make the minimum go version to be 1.16

File: template_loader.go [hunk 2]
```diff
@@ -211,3 +212,56 @@ func (h *HttpFilesystemLoader) Get(path string) (io.Reader, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
```
Comment (cam-stitt) [inline template_loader.go:259]: Although closing the files is probably best practice, the embedded filesystem does nothing when `Close` is called. We could potentially save quite a bit of performance overhead here, as every file in an embedded filesystem is already loaded and will be handled through garbage collection.

https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/embed/embed.go;l=342;drc=refs%2Ftags%2Fgo1.17.7

Comment (fahmifan) [reply template_loader.go:259]: Cool, so we can remove the line, right?

Comment (cam-stitt) [reply template_loader.go]: I was more thinking there is no need to both open the file, read it and place the contents into a buffer, we should be able to just return the `e.fs.Open` call.

File: template_loader.go [hunk 3]
```diff
@@ -211,3 +212,56 @@ func (h *HttpFilesystemLoader) Get(path string) (io.Reader, error) {
 
 	return h.fs.Open(fullPath)
 }
+
+// EmbededLoader support loading templates from embed.FS
+// embed.FS file names pattern are limited check https://pkg.go.dev/embed#hdr-Directives
+// and https://github.com/golang/go/issues/43854
+type EmbededLoader struct {
```
Comment (kevlened) [inline template_loader.go:219]: I'm excited to see progress on this PR! Small nit: `EmbededLoader` should be `EmbeddedLoader`.

**Comment by flosch** [comment]
I already added FSLoader supporting fs.FS to pongo v5. Thanks.
